src/main/java/client/Character.java

```diff
@@ -323,6 +325,45 @@ public void onAnnounceStatPoolUpdate() {
         setPosition(new Point(0, 0));
     }
 
+    static {
+        if (YamlConfig.config.server.USE_EXP_GAIN_LOG) {
+            SaveExpLogToDBThread expThread = new SaveExpLogToDBThread();
+            expThread.setPriority(Thread.MIN_PRIORITY); // Set to the lowest priority
+            expThread.start();
+        }
+    }
+
+    // Nested static class for saving exp logs to the database
+    private static class SaveExpLogToDBThread extends Thread {
+        @Override
+        public void run() {
+            while(true) {
+                try {
+                    if (expGainLogQueue.isEmpty()) {
+                        synchronized (expGainLogQueue) {
+                            expGainLogQueue.wait();
+                        }
+                    }
+                    try (Connection con = DatabaseConnection.getConnection();
+                        PreparedStatement ps = con.prepareStatement("INSERT INTO characterexplogs (world_exp_rate, exp_coupon, gained_exp, current_exp, exp_gain_time, charid) VALUES (?, ?, ?, ?, ?, ?)")) {
+                        while (!expGainLogQueue.isEmpty()) {
+                            Map<String, Object> psMap = expGainLogQueue.poll();
+                            ps.setInt(1, (int) psMap.get("world_exp_rate"));
+                            ps.setInt(2, (int) psMap.get("exp_coupon"));
+                            ps.setLong(3,(long) psMap.get("gained_exp"));
+                            ps.setInt(4, (int) psMap.get("current_exp"));
+                            ps.setTimestamp(5, (Timestamp) psMap.get("exp_gain_time"));
+                            ps.setInt(6, (int) psMap.get("charid"));
+                            ps.executeUpdate();
+                        }
+                    }
+                } catch (SQLException | InterruptedException e) {
+                    e.printStackTrace();
+                }
+            }
+        }
+    }
```

Comment:
Character.java is big enough already. Should split this off to a separate class, no reason to keep it in here.